client/app/src/main/java/com/example/criengine/Activities/AddBookActivity.java

```diff
@@ -90,6 +90,7 @@ public void onClick(View v) {
             @Override
             public void onClick(View v) {
                 Intent intent = new Intent(v.getContext(), MyBooksActivity.class);
+                intent.putExtra("Index", RootActivity.PAGE.MY_BOOKS);
```

Comment:
If you're doing this here, you may as well override the back button as well.
```
    /**
     * Overrides the back button so it returns to the main activity.
     */
    @Override
    public void onBackPressed() {
        Intent intent = new Intent(this, RootActivity.class);
        intent.putExtra("Index", 3);
        startActivity(intent);
```
Copy paste that into the bottom of this class.

Reply:
i don't think it should since this is the activity, it's not necessarily part of the navbar stuff

Reply:
The back button gets pretty weird when you allow it to return to the literal last screen. Doing this can avoid future issues